src/lib/y2storage/dialogs/guided_setup/select_scheme.rb

```diff
@@ -0,0 +1,170 @@
+                  Password(Id(:repeat_password), _("Verify Password"))
+                )
+              )
+            )
+          )
+        end
+
+        def initialize_widgets
+          widget_update(:lvm, settings.use_lvm)
+          widget_update(:encryption, settings.use_encryption)
+          encryption_handler
+          if settings.use_encryption
+            widget_update(:password, settings.encryption_password)
+            widget_update(:repeat_password, settings.encryption_password)
+          end
+        end
+
+        def update_settings!
+          valid = valid_settings?
+          if valid
+            settings.use_lvm = widget_value(:lvm)
+            password = widget_value(:password)
+            settings.encryption_password = password unless password.to_s.empty?
+          end
+          valid
+        end
+
+      private
+
+        PASS_MIN_SIZE = 5
+
+        PASS_ALLOWED_CHARS =
+          "0123456789" \
+          "abcdefghijklmnopqrstuvwxyz" \
+          "ABCDEFGHIJKLMNOPQRSTUVWXYZ" \
+          "#* ,.;:._-+=!$%&/|?{[()]}@^\\<>"
+
+        def valid_settings?
+          using_encryption? ? valid_password? : true
+        end
+
+        def using_encryption?
+          widget_value(:encryption)
+        end
+
+        def valid_password?
+          !password_blank? && password_match? &&
+            password_correct? && password_strong?
+        end
+
+        def password_blank?
+          return false unless widget_value(:password).empty?
+          Yast::Report.Warning(_("A password is needed"))
+          true
+        end
+
+        def password_match?
+          return true if widget_value(:password) == widget_value(:repeat_password)
+          Yast::Report.Warning(_("Password does not match"))
+          false
+        end
+
+        def password_correct?
+          correct = password_min_size? && password_allowed_chars?
+
+          if !correct
+            messages = [
+              _("The password must have at least #{PASS_MIN_SIZE} characters."),
+              _("The password may only contain the following characters:\n" \
+                "0..9, a..z, A..Z, and any of \"@#* ,.;:._-+=!$%&/|?{[()]}^\\<>\".")
+            ]
+            Yast::Report.Warning(messages.join("\n"))
+          end
+
+          correct
+        end
+
+        def password_strong?
+          message = crack_password
+          return true if message.nil? # cracklib fails
+          return true if message.empty?
+          Yast::Report.Warning(message)
+          false
+        end
+
+        def password_min_size?
+          password.size >= PASS_MIN_SIZE
+        end
+
+        def password_allowed_chars?
+          password.split(//).all? { |c| PASS_ALLOWED_CHARS.include?(c) }
+        end
+
+        # Check password strength using cracklib.
+        # @return[String] crack lib message, empty ("") if successful.
+        def crack_password
+          Yast::InstExtensionImage.with_extension("cracklib-dict-full.rpm") do
```

Comment:
Already commented offline. Loading and unloading the extension of every check is too much. Unload it only when leaving the dialog.